internal/dao/pod.go

```diff
@@ -240,20 +240,64 @@ func tailLogs(ctx context.Context, logger Logger, c chan<- []byte, opts LogOptio
 	req.Context(ctx)
 
 	var blocked int32 = 1
+	var emptyLogs int32 = 0
+
+	logsCheckerCtx, logsCheckerCancel := context.WithCancel(ctx)
+	go logsEmptyChecker(logsCheckerCtx, cancel, logger, opts, &blocked, &emptyLogs)
 	go logsTimeout(cancel, &blocked)
 
 	// This call will block if nothing is in the stream!!
+	// That's why we doing parallel request above which will
+	// cancel this stream in case logs size is zero
+	// But we anyways using timeout in both contexts in case of connection issues
```

Comment:
But anyway, we are using the same timeout in both contexts for the case of connection issues